pallets/bridge/src/lib.rs

```diff
@@ -316,8 +319,24 @@ decl_storage! {
 
         /// Details of bridge transactions identified with pairs of the recipient account and the
         /// bridge transaction nonce.
-        BridgeTxDetails get(fn bridge_tx_details):
-            double_map
+        BridgeTxDetails get(fn bridge_tx_details) build(|config: &GenesisConfig<T>| {
+            // Record the transactions in genesis.
+            config.complete_txs.iter().map(|tx| {
+                let detail = BridgeTxDetail {
+                    amount: tx.amount,
+                    status: BridgeTxStatus::Handled,
+                    execution_block: Zero::zero(),
+                    tx_hash: tx.tx_hash,
+                };
+                debug::info!(
+                    "Genesis bridge transaction to {:?} with nonce {} for {:?} POLYX",
+                    tx.recipient,
+                    tx.nonce,
+                    tx.amount
+                );
+                (tx.recipient.clone(), tx.nonce, detail)
+            }).collect::<Vec<_>>()
```

Comment:
Could you add an integration test for this on-genesis logic?

Reply:
Yes, I'm working on it.